test/StripeResource.spec.js

```diff
@@ -208,6 +216,98 @@ describe('StripeResource', () => {
         );
       });
 
+      it('throws an error on invalid JSON', (done) => {
```

Comment:
Would there be anyway to add a test for when there's a connection error *after* the headers have been transmitted, but halfway through when the body is being transmitted?

(this should be pretty similar to the test case above on line #200).

That's the one case I can think of where (correct me if I'm wrong) I don't think we have any test coverage but it still would be good to know that this PR is consistent with the previous behavior (and that `fetch` and `nodehttpclient` will behave consistently, too)

Reply:
Done.